src/Kestrel.Tls/TlsConnectionAdapter.cs

```diff
@@ -47,6 +47,17 @@ public TlsConnectionAdapter(TlsConnectionAdapterOptions options, ILoggerFactory
 
             _options = options;
             _logger = loggerFactory?.CreateLogger(nameof(TlsConnectionAdapter));
+
+            // Order is important. If HTTP/2 is enabled, we prefer it over HTTP/1.1. So add it first.
+            if ((options.Protocols & HttpProtocols.Http2) == HttpProtocols.Http2)
```

Comment:
You could #ifdef this section in preparation for the new core API. Then you'd be able to test the net46 behavior.

Reply:
Since I can't properly test with ALPN now, I'll do that in a future change.